proxy-node-gateway/src/main/java/uk/gov/ida/notification/redis/storage/StorageType.java

```diff
@@ -0,0 +1,12 @@
+package uk.gov.ida.notification.redis.storage;
+
+import uk.gov.ida.notification.session.GatewaySessionData;
+
+public interface StorageType {
+
+    void addSession(String sessionId, GatewaySessionData sessionData);
+    boolean sessionExists(String sessionId);
+    GatewaySessionData getSession(String sessionId);
+    void start();
```

Comment:
These are just being handed straight to `Managed` - could we just make this interface `extend Managed` to make it clear that anything that implements this _has_ to be `Managed` but the fundamental behaviour of this interface is about sessions, not stopping and starting?